pkg/disk/partition_table.go

```diff
@@ -678,6 +689,62 @@ func (pt *PartitionTable) ensureLVM() error {
 	return nil
 }
 
+// ensureBtrfs will ensure that the root partition is on a btrfs subvolume, i.e. if
+// it currently is not, it will wrap it in one
+func (pt *PartitionTable) ensureBtrfs() error {
+
+	rootPath := entityPath(pt, "/")
+	if rootPath == nil {
+		panic("no root mountpoint for PartitionTable")
+	}
+
+	// we need a /boot partition to boot btrfs, ensure one exists
+	bootPath := entityPath(pt, "/boot")
+	if bootPath == nil {
+		_, err := pt.CreateMountpoint("/boot", 512*common.MiB)
+
+		if err != nil {
+			return err
+		}
+
+		rootPath = entityPath(pt, "/")
```

Comment:
Silly question (again) - why do we set the root path here again, didn't we do this in line 696 already? The two calls look identical to me? Or does the mutation of `pt.CreateMountPoint()` above require that rootPath is recalculated?